pkg/template/resource.go

```diff
@@ -117,3 +97,31 @@ func Uid() string {
 func ApplyUIDToResourceTemplate(rt json.RawMessage, uid string) json.RawMessage {
 	return bytes.Replace(rt, uidMatch, []byte(uid), -1)
 }
+
+// MergeParams merges two param arrays. An error is returned if there are
+// multiple params with the same name.
+func MergeParams(params1 []pipelinev1.Param, params2 []pipelinev1.Param) ([]pipelinev1.Param, error) {
```

Comment:
IIUC, this is the same as `MergeInDefaultParams()`, but does not allow for duplicates. These could be combined (maybe), but I think they are distinct enough as well.

Reply:
`MergeParams` merges two lists of `Param`s, whereas `MergeInDefaultParams` takes one list of `Param`s, and one list of `ParamSpec`s, then adds the `default` values for params not provided in the list

Reply:
Do you think there's a more descriptive name that I can use to make this distinction more clear?